sfputil/main.py

```diff
@@ -1564,5 +1603,186 @@ def target(port_name, target):
         click.echo("Target Mode set failed!")
         sys.exit(EXIT_FAIL)
 
+
+# 'read-eeprom' subcommand
+@cli.command()
+@click.argument('port_name', metavar='<port_name>', required=True)
+@click.argument('page', metavar='<page>', type=click.INT, required=True)
+@click.argument('offset', metavar='<offset>', type=click.INT, required=True)
+@click.argument('size', metavar='<size>', type=click.INT, required=True)
+@click.option('--no-format', is_flag=True, help="Display non formatted data")
+@click.option('--wire-addr', help="Wire address of sff8472")
+def read_eeprom(port_name, page, offset, size, no_format, wire_addr):
+    """Read SFP EEPROM data"""
+    try:
+        if platform_sfputil.is_logical_port(port_name) == 0:
+            click.echo("Error: invalid port {}".format(port_name))
+            print_all_valid_port_values()
+            sys.exit(ERROR_INVALID_PORT)
+
+        if is_port_type_rj45(port_name):
+            click.echo("This functionality is not applicable for RJ45 port {}.".format(port_name))
+            sys.exit(EXIT_FAIL)
+
+        physical_port = logical_port_to_physical_port_index(port_name)
+        sfp = platform_chassis.get_sfp(physical_port)
+        if not sfp.get_presence():
+            click.echo("{}: SFP EEPROM not detected\n".format(port_name))
+            sys.exit(EXIT_FAIL)
+
+        from sonic_platform_base.sonic_xcvr.api.public import sff8472
+        api = sfp.get_xcvr_api()
+        if api is None:
+            click.echo('Error: SFP EEPROM not detected!')
+        if not isinstance(api, sff8472.Sff8472Api):
+            overall_offset = get_overall_offset_general(api, page, offset, size)
+        else:
+            overall_offset = get_overall_offset_sff8472(api, page, offset, size, wire_addr)
+        return_code, output = eeprom_dump_general(physical_port, page, overall_offset, size, offset, no_format)
+        if return_code != 0:
+            click.echo("Error: Failed to read EEPROM!")
+            sys.exit(return_code)
+        click.echo(output)
+    except NotImplementedError:
+        click.echo("This functionality is currently not implemented for this platform")
+        sys.exit(ERROR_NOT_IMPLEMENTED)
+    except ValueError as e:
+        click.echo(f"Error: {e}")
+        sys.exit(EXIT_FAIL)
+
+
+# 'write-eeprom' subcommand
+@cli.command()
+@click.argument('port_name', metavar='<port_name>', required=True)
+@click.argument('page', metavar='<page>', type=click.INT, required=True)
+@click.argument('offset', metavar='<offset>', type=click.INT, required=True)
+@click.argument('data', metavar='<data>', required=True)
+@click.option('--wire-addr', help="Wire address of sff8472")
+@click.option('--verify', is_flag=True, help="Verify the data by reading back")
```

Comment:
@Junchao-Mellanox Click has various argument like check the range. https://click.palletsprojects.com/en/7.x/api/#click.IntRange

This avoid duplicate code check for the same